SIPS/sip-seth-lp-rewards.md

```diff
@@ -0,0 +1,77 @@
+---
+sip: 31
+title: sETH LP rewards contract
+status: WIP
+author: Clinton Ennis (@hav-noms), Anton Bukov (@k06a) 
+discussions-to: https://discord.gg/3uJ5rAy
+
+created: 2019-11-04
+---
+
+<!--You can leave these HTML comments in your merged SIP and delete the visible duplicate text guides, they will not appear and may be helpful to refer to if you edit it again. This is the suggested template for new SIPs. Note that an SIP number will be assigned by an editor. When opening a pull request to submit your SIP, please use an abbreviated title in the filename, `sip-draft_title_abbrev.md`. The title should be 44 characters or less.-->
+
+## Simple Summary
+
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the SIP.-->
+
+Replace the current uniswap sETH Liquidity Provider mechanism with an onchain LP rewards staking contract.
+
+## Abstract
+
+<!--A short (~200 word) description of the technical issue being addressed.-->
+Note: LP = Liquidity Provider
+
+The current deployed solution is inneficient and buggy. There are weekly issues with valid LP's not getting rewards and cheaters getting rewards that should be disqualified. 
+This onchain solution fixes all the issues and can be automated and self serve requiring much less steps than the current solution. 
+
+## Motivation
+
+<!--The motivation is critical for SIPs that want to change Synthetix. It should clearly explain why the existing protocol specification is inadequate to address the problem that the SIP solves. SIP submissions without sufficient motivation may be rejected outright.-->
+
+- Current solution uses an off-chain [python script](http://18.222.88.2:5000/pool-rewards/8926035/8967962) to determine sETH Liquidity providers between two blocks which are manually entered
+- We must download a CSV and manually verify the addresses and their LP amounts
+- There are so many LPs now that it takes 2 multi-sig multi-send transactions to send out the LP SNX rewards
+- The co-ordination amongst the 5 multi-sigs signers and overhead each manually verifying is inefficient given this could all be offloaded to a trusted smart contract to calculate on-chain in a trust-less manor. 
+- The [gnosis multi-sig](https://wallet.gnosis.pm/#/wallet/0x53265D3D34c9ECB5685Be3176430366b4e392010) freezes and is very slow to react since the payloads of the transactions are so big 
+
```

Comment:
@hav-noms Another frustration LP providers have noted is that if they withdraw / rebalance any amount of their liquidity from the pool during the week then all their rewards will be forfeited. The python script checks that no withdrawals / transfer of LP tokens have been made from the pool to prevent people from gaming the rewards.